spec/components/radio_spec.rb

```diff
@@ -149,6 +149,23 @@ def component_name
     assert_select "legend span.govuk-caption-xl", false
   end
 
+  it "renders radio-group with a custom page heading size" do
+    render_component(
+      name: "favourite-skittle",
+      heading: "What is your favourite skittle?",
+      heading_caption: "Question 3 of 9",
+      heading_size: "l",
+      is_page_heading: true,
+      items: [
+        { label: "Red", value: "red" },
+        { label: "Blue", value: "blue" },
+      ],
+    )
+    assert_select ".govuk-radios"
+    assert_select "legend.govuk-fieldset__legend--l h1.govuk-fieldset__heading", text: "What is your favourite skittle?"
+    assert_select "legend span.govuk-caption-l", "Question 3 of 9"
+  end
+
```

Comment:
I think we might need a few more tests, correct me if I'm wrong.

- a test that `size` defaults to `m` if no `heading_size` is passed and `is_page_heading` is not set
- a test that `size` is `xl` if `is_page_heading` is true

Reply:
Thanks for adding some more tests, this looks much better.